skrifa/src/outline/glyf/hint/call_stack.rs

```diff
@@ -0,0 +1,92 @@
+//! Tracking function call state.
+
+use super::{definition::Definition, error::HintErrorKind, program::Program};
+
+// FreeType provides a call stack with a depth of 32.
+// See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L502>
+const MAX_DEPTH: usize = 32;
+
+/// Record of an active invocation of a function or instruction
+/// definition.
+///
+/// See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.h#L90>
+#[derive(Copy, Clone, Default)]
+pub struct CallRecord {
+    pub caller_program: Program,
+    pub return_pc: usize,
+    pub current_count: u32,
+    pub definition: Definition,
+}
+
+/// Tracker for nested active function or instruction calls.
+#[derive(Default)]
+pub struct CallStack {
+    records: [CallRecord; MAX_DEPTH],
+    top: usize,
```

Comment:
imho the behavior of this would be more intuitive if it was called "len" which more directly implies 0..len (non-inclusive) are the valid indices

Reply:
Done